source/docs/software/advanced-controls/trajectories/constraints.rst

```diff
@@ -0,0 +1,58 @@
+Trajectory Constraints
+======================
+In the :ref:`previous article <docs/software/advanced-controls/trajectories/trajectory-generation:Generating the trajectory>`, you might have noticed that no custom constraints were added when generating the trajectories. Custom constraints allow users to impose more restrictions on the velocity and acceleration at points along the trajectory based on location and curvature.
+
+For example, a custom constraint can keep the velocity of the trajectory under a certain threshold in a certain region or slow down the robot near turns for stability purposes.
+
+WPILib-Provided Constraints
+---------------------------
+WPILib includes a set of predefined constraints that users can utilize when generating trajectories. The list of WPILib-provided constraints is as follows:
+
+ - ``CentripetalAccelerationConstraint``: Limits the centripetal acceleration of the robot as it traverses along the trajectory. This can help slow down the robot around right turns.
+ - ``DifferentialDriveKinematicsConstraint``: Limits the velocity of the robot around turns such that no wheel of a differential-drive robot goes over a specified maximum velocity.
+ - ``DifferentialDriveVoltageConstraint``: Limits the acceleration of a differential-drive robot such that no commanded voltage goes over a specified maximum.
```

Comment:
```suggestion
 - ``DifferentialDriveVoltageConstraint``: Limits the acceleration of a differential drive robot such that no commanded voltage goes over a specified maximum.
```
This description is confusing, but I'm not sure there's much we can do about it. In my opinion, it's easier to think of this in terms of dynamics rather than voltage (and https://github.com/wpilibsuite/allwpilib/pull/2364 shows this formulation is simpler). If we were to do it again, I'd call it DifferentialDriveDynamicsConstraint and say "Limits the acceleration of the differential drive according to the provided velocity dynamics and maximum voltage."